### PR TITLE
Try using routes DSL

### PR DESCRIPTION
**Pull request by rylev**
This isn't cleaned up 100% but what do you think about the direction? @myobie  @duncan

### CODE REVIEW

File: src/main/scala/Router.scala [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  def resources(resource: Resource) = {
+    pathPrefix("api") {
+      pathPrefix("v1") {
+        path(resource.name) {
+          get { resource.index() } ~
+          post { resource.create() }
+        } ~
+        path(resource.name / IntNumber) { id =>
+          get { resource.show() } ~
+          (put | patch) { resource.update() } ~
```
Comment (ghost) [inline src/main/scala/Router.scala]: we should define a val that is the combination of `(put | patch)` I vote for `patut` (j/k)

**Comment by duncan** [comment]
On the one hand, I really like the matches returning futures that was in the previous. On the other, the sugar provided by the DSL is nice. My thought with this is comes to the question of how Wunderlisty this example should be. Should it assume it's providing a resource under `/api/v1` etc? OTOH, if you're going to go as far as providing that and a model controller like `users`, then should versioning be baked in somehow? But then that definitely feels like it's going to far for a basic example.

**Comment by duncan** [comment]
Sorry, didn't mean to close! :)

**Comment by atdrendel** [comment]
Closing this since it was last active in 2015.

**Pull request closed without merging**